src/services/smartWallet.js

```diff
@@ -313,16 +332,26 @@ class SmartWallet {
 
   async estimateAccountDeployment(gasInfo: GasInfo) {
     const deployEstimate = await this.sdk.estimateAccountDeployment().catch(() => {});
-    let { gasAmount, gasPrice } = parseEstimatePayload(deployEstimate);
+    return calculateEstimate(deployEstimate, gasInfo, SPEED_TYPES.FAST, 790000);
+  }
 
-    if (!gasAmount) {
-      gasAmount = new BigNumber(790000);
-    }
-    if (!gasPrice) {
-      const defaultGasPrice = get(gasInfo, 'gasPrice.max', 0);
-      gasPrice = utils.parseUnits(defaultGasPrice.toString(), 'gwei');
-    }
-    return gasPrice.mul(gasAmount);
+  async estimateAccountTransaction(transaction: AccountTransaction, gasInfo: GasInfo) {
+    const {
+      recipient,
+      value,
+      data,
+      transactionSpeed = TransactionSpeeds[AVG],
+    } = transaction;
+    const estimatedTransaction = await this.sdk.estimateAccountTransaction(
+      recipient,
+      value,
+      data,
+      transactionSpeed,
+    ).catch(() => {});
```

Comment:
It looks like `calculateEstimate` should receive an `EstimatePayload`, maybe we can return if we know the sdk call failed instead of passing an empty object.

Reply:
Well if if fails then it will return default values as defined on `calculateEstimate`.

Reply:
Yeah, but this way we can't add a type to the `estimate` parameter on `calculateEstimate`, right?